server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1624,30 +1602,13 @@ public Entitlement adjustEntitlementQuantity(Consumer consumer, Entitlement enti
         Map<String, Integer> poolQuantityMap, CallerType caller)
         throws EntitlementRefusedException {
 
-        BindChain chain = new BindChain();
-        BindContext context = bindContextFactory.create(consumer, poolQuantityMap);
-        chain.addOperation(rulesCheckOpFactory.create(caller));
-        chain.addOperation(handleEntitlementsOp);
-        chain.addOperation(postBindBonusPoolsOp);
-        chain.addOperation(handleCertificatesOp);
-        chain.addOperation(complianceOp);
-
-        // fetch and pre-process entities before locks are placed.
-        chain.preProcess(context);
-        if (context.getException() == null) {
-            chain.lock(context);
-            // post lock operations
-            chain.execute(context);
-        }
-
-        //check again cause either pre process or execute can throw an exception.
-        if (context.getException() != null) {
-            throw context.getException();
-        }
+        Map<String, Entitlement> result = bindChainFactory.
+            create(consumer, poolQuantityMap, caller).
+            run();
```

Comment:
If you are just going to call result.values() below, any reason to return Map<String, Entitlement> instead of just a list?